docs/DeveloperGuide.md

```diff
@@ -95,13 +95,15 @@ Here's a (partial) class diagram of the `Logic` component:
 
 How the `Logic` component works:
 1. When `Logic` is called upon to execute a command, it uses the `AddressBookParser` class to parse the user command.
-1. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `CreateCommand`) which is executed by the `LogicManager`.
-1. The command can communicate with the `Model` when it is executed (e.g. to add a person).
-1. The result of the command execution is encapsulated as a `CommandResult` object which is returned back from `Logic`.
+2. This results in a `Command` object (more precisely, an object of one of its subclasses e.g., `CreateCommand`) which is executed by the `LogicManager`.
+   1. `UndoableCommand` is another interface that extends from the `Command` interface which represents commands that have undo/redo logic.
+   2. Commands that implements `UndoableCommand` are managed in the `CommandManager`.
+3. The command can communicate with the `Model` when it is executed (e.g. to add a person).
+4. The result of the command execution is encapsulated as a `CommandResult` object which is returned back from `Logic`.
```

Comment:
i think shld not use the word encapsulate? may sound like the wrong oop jargon? maybe sth like the return type of the command execution is an command result object